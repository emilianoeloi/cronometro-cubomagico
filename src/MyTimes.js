import React, { Component } from 'react';

import { msToISOString } from './Common';

class MyTimes extends Component {
  render() {
    const {
      bestTime,
      mediumTime,
      wrostTime,
      times,
    } = this.props;
    if (!times) {
      return null;
    }
    let i = 1;
    return (
      <table className="pure-table">
        <caption>
          <h2>Meus Tempos</h2>
          <h3>Melhor tempo: {msToISOString(bestTime)}</h3>
          <h3>Tempo m√©dio: {msToISOString(mediumTime)}</h3>
          <h3>Pior tempo: {msToISOString(wrostTime)}</h3>
        </caption>
        <thead>
            <tr>
                <th>#</th>
                <th>Data</th>
                <th>Tempo</th>
            </tr>
        </thead>

        <tbody>
          {times.map(time => (
            <tr key={time.key} className={i%2 === 0 ? 'pure-table-odd' : ''}>
                <td>{i++}</td>
                <td>{new Date(time.date).toLocaleDateString()}</td>
                <td>{msToISOString(time.time)}</td>
            </tr>
          ))}
        </tbody>
      </table>
    )
  }
}

export default MyTimes;
