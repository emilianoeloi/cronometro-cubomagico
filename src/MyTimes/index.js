import React, { Component } from 'react';

import { msToISOString } from '../Common';

class MyTimes extends Component {
  constructor(props) {
    super(props);
    this.removeTime = this.removeTime.bind(this);
  }
  removeTime(element) {
    const key = element.target.dataset['key'];
    this.props.removeTime(key);
  }
  render() {
    const {
      bestTime,
      mediumTime,
      wrostTime,
      times,
      removeTime,
    } = this.props;
    if (!times) {
      return null;
    }
    let i = 1;
    return (
      <table className="pure-table">
        <caption className="Table-caption">
          <h2>Meus Tempos</h2>
          <h3>Melhor tempo: {msToISOString(bestTime)}</h3>
          <h3>Tempo m√©dio: {msToISOString(mediumTime)}</h3>
          <h3>Pior tempo: {msToISOString(wrostTime)}</h3>
        </caption>
        <thead>
          <tr>
            <th>#</th>
            <th>Data</th>
            <th>Tempo</th>
            <th></th>
          </tr>
        </thead>

        <tbody>
          {times.map(time => (
            <tr key={time.key} className={i%2 === 0 ? 'pure-table-odd' : ''}>
              <td>{i++}</td>
              <td>{new Date(time.date).toLocaleDateString()}</td>
              <td className="time-formated">{msToISOString(time.time)}</td>
              <td>
                <button data-key={time.key} onClick={this.removeTime} className="button-delete pure-button">
                  excluir
                </button>
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    )
  }
}

export default MyTimes;
