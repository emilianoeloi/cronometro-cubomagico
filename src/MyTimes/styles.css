.button-delete {
  background: rgb(202, 60, 60); /* this is a maroon */
  color: #fff;
  font-size: 70%;
}
