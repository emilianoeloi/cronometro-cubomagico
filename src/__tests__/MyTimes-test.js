import React from 'react';
import renderer from 'react-test-renderer';

import MyTimes from '../MyTimes';

test('test table of Times', () => {
  const component = renderer.create(
    <MyTimes />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
