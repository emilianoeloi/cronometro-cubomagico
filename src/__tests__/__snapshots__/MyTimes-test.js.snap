exports[`test test table of Times 1`] = `
<table
  className="pure-table">
  <caption
    className="Table-caption">
    <h2>
      Meus Tempos
    </h2>
    <h3>
      Melhor tempo: 
      00\' 00.00\"
    </h3>
    <h3>
      Tempo m√©dio: 
      00\' 00.00\"
    </h3>
    <h3>
      Pior tempo: 
      00\' 00.00\"
    </h3>
  </caption>
  <thead>
    <tr>
      <th>
        #
      </th>
      <th>
        Data
      </th>
      <th>
        Tempo
      </th>
      <th />
    </tr>
  </thead>
  <tbody />
</table>
`;
